2024/day16/day16.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	_ "embed"
+	"fmt"
+	"math"
+	"sort"
+	"strings"
+)
+
+//go:embed test.txt
+var test string
+
+//go:embed input.txt
+var input string
+
+func main() {
+	input := ParseInput(input)
+
+	for i := 0; i < len(input); i++ {
+		for j := 0; j < len(input[0]); j++ {
+			if input[i][j] == "S" {
+				fmt.Println("Starting", i, j)
+				out := FindPath(i, j, input)
+				fmt.Println(out)
+				break
+			}
+		}
+	}
+}
+
+func FindPath(sx, sy int, board [][]string) int {
+	min_v := math.MaxInt64
+	seen := make(map[[2]int]bool)
+
+	directions := []struct {
+		dx, dy int
+		dir    string
+	}{
+		{1, 0, "S"},  // South
+		{-1, 0, "N"}, // North
+		{0, 1, "E"},  // East
+		{0, -1, "W"}, // West
+	}
+
+	queue := [][4]interface{}{{sx, sy, 1001, ""}}
+	seen[[2]int{sx, sy}] = true
+
+	for len(queue) > 0 {
+		sort.SliceStable(queue, func(i, j int) bool {
+			return queue[i][2].(int) < queue[j][2].(int)
+		})
```

Comment:
Interesting Explanation for Dijkstra Here
https://stackoverflow.com/questions/30409493/using-bfs-for-weighted-graphs
@BhavyaMuni